src/_guides/language/numbers.md

```diff
@@ -0,0 +1,547 @@
+   <td><code>2147483647</code></td>
+   <td><code>2147483647</code></td>
+  </tr>
+</table>
+
+
+### String representation
+
+On the web, Dart generally defers to JavaScript to convert a number to a string
+(for example, for a `print`).
+The following table demonstrates how
+converting the expressions in the first column can lead to different results.
+
+<table class="table table-striped nowrap">
+  <tr>
+   <th>Expression</th>
+   <th>Native <code>toString()</code></th>
+   <th>Web <code>toString()</code></th>
+  </tr>
+  <tr>
+   <td><code>1</code></td>
+   <td><code>"1"</code></td>
+   <td><code>"1"</code></td>
+  </tr>
+  <tr>
+   <td><code>1.0</code></td>
+   <td><code>"1.0"</code></td>
+   <td><code>"1"</code></td>
+  </tr>
+  <tr>
+   <td><code>(0.5 + 0.5)</code></td>
+   <td><code>"1.0"</code></td>
+   <td><code>"1"</code></td>
+  </tr>
+  <tr>
+   <td><code>1.5</code></td>
+   <td><code>"1.5"</code></td>
+   <td><code>"1.5"</code></td>
+  </tr>
+  <tr>
+   <td><code>-0</code></td>
+   <td><code>"0"</code></td>
+   <td><code>"-0.0"</code></td>
+  </tr>
+  <tr>
+   <td><code>math.pow(2, 0)</code></td>
+   <td><code>"1"</code></td>
+   <td><code>"1"</code></td>
+  </tr>
+  <tr>
+   <td><code>math.pow(2, 80)</code></td>
+   <td><code>"0"</code></td>
+   <td><code>"1.2089258196146292e+24"</code></td>
+  </tr>
+</table>
+
+
+## What should you do?
+
+Usually, you don't need to do anything.
+Dart code has been running on both native and web platforms for years,
+and number implementation differences are rarely a problem.
+Common, typical code —
+such as iterating through a range of small integers and indexing a list —
+behaves the same.
+
+If you have tests or assertions that compare string results,
+write them in a platform-resilient manner.
+For example, suppose you're testing the value of string expressions
+that have embedded numbers:
+
+```dart
+void main() {
+  var count = 10.0 * 2;
+  var message = "$count cows";
+  if (message != "20.0 cows") throw Exception("Unexpected: $message");
+}
+```
+
+The preceding code succeeds on native platforms but throws on the web
+because `message` is `"20 cows"` (no decimal) on the web.
+As an alternative, you might write the condition as follows,
+so it passes on both native and web platforms:
+
+```dart
+if (message != "${20.0} cows") throw ... 
+```
+
+For bit manipulation, consider explicitly operating on 32-bit chunks,
+which are consistent on all platforms.
+To force a signed interpretation of a 32-bit chunk,
+use `int.toSigned(32)`.
+
+For other cases where precision matters,
+consider other numeric types.
+The [`BigInt`][] type
+provides arbitrary-precision integers on both native and web.
+The [`fixnum`][] package
+provides strict 64-bit signed numbers, even on the web.
+Use these types with care, though:
+they often result in significantly bigger and slower code.
+
+[`BigInt`]: https://api.dart.dev/dev/2.12.0-281.0.dev/dart-core/BigInt-class.html
```

Comment:
Can you switch this link to a non-version specific one?

https://api.dart.dev/dart-core/BigInt-class.html

Reply:
Oops, I meant to do that. Good catch!